xchange-okcoin/src/main/java/com/xeiam/xchange/okcoin/service/streaming/OkCoinBaseStreamingService.java

```diff
@@ -5,42 +5,40 @@
 
 import org.java_websocket.WebSocket.READYSTATE;
 
+import com.xeiam.xchange.Exchange;
 import com.xeiam.xchange.ExchangeSpecification;
-import com.xeiam.xchange.currency.CurrencyPair;
 import com.xeiam.xchange.okcoin.OkCoinExchange;
 import com.xeiam.xchange.service.streaming.ExchangeEvent;
 import com.xeiam.xchange.service.streaming.ExchangeStreamingConfiguration;
 import com.xeiam.xchange.service.streaming.StreamingExchangeService;
 
-public class OkCoinStreamingExchangeService implements StreamingExchangeService {
-  private final WebSocketBase socketBase;
-  private final BlockingQueue<ExchangeEvent> eventQueue = new LinkedBlockingQueue<ExchangeEvent>();
-  private final OkCoinExchangeStreamingConfiguration exchangeStreamingConfiguration;
-  private final ChannelProvider channelProvider;
+public class OkCoinBaseStreamingService implements StreamingExchangeService {
+  private WebSocketBase socketBase;
+  protected final ChannelProvider channelProvider;
+  private final BlockingQueue<ExchangeEvent> marketDataEventQueue = new LinkedBlockingQueue<ExchangeEvent>();
 
-  public OkCoinStreamingExchangeService(ExchangeSpecification exchangeSpecification, ExchangeStreamingConfiguration exchangeStreamingConfiguration) {
-    this.exchangeStreamingConfiguration = (OkCoinExchangeStreamingConfiguration) exchangeStreamingConfiguration;
+  private final OkCoinStreamingConfiguration exchangeStreamingConfiguration;
 
+  public OkCoinBaseStreamingService(Exchange exchange, ExchangeStreamingConfiguration streamingConfiguration) {
+    
+    this.exchangeStreamingConfiguration = (OkCoinStreamingConfiguration) streamingConfiguration;
+
+    ExchangeSpecification exchangeSpecification = exchange.getExchangeSpecification();
     String sslUri = (String) exchangeSpecification.getExchangeSpecificParametersItem("Websocket_SslUri");
     boolean useFutures = (Boolean) exchangeSpecification.getExchangeSpecificParametersItem("Use_Futures");
 
     channelProvider = useFutures ? new FuturesChannelProvider(OkCoinExchange.futuresContractOfConfig(exchangeSpecification))
-        : new SpotChannelProvider();
-
-    WebSocketService socketService = new OkCoinWebSocketService(eventQueue, channelProvider,
+        : new SpotChannelProvider(exchangeSpecification);
+    WebSocketService socketService = new OkCoinWebSocketService(marketDataEventQueue,
+        channelProvider,
         this.exchangeStreamingConfiguration.getMarketDataCurrencyPairs());
+    
     socketBase = new WebSocketBase(sslUri, socketService);
   }
 
   @Override
   public void connect() {
     socketBase.start();
-
```

Comment:
I would move these lines to connect() method of the class which is currently called OkCoinStreamingMarketDataService (and then would rename it to OkCoinStreamingExchangeService to ensure backword compatibility)